### ISSUE TITLE
Rake tasks shouldn't trigger Rollbar errors

### ISSUE THREAD

**Issue by sman591** [codeRIT/brickhack.io#212]

**Comment by ztmeyer** [comment]
Is this still an issue? If so is it occurring in production only?

I see this in `rollbar.rb`:
```ruby
  # Here we'll disable Rollbar when not in 'production':
  config.enabled = false unless Rails.env.production?
```

**Comment by sman591** [comment]
@ztmeyer correct - production only (as set by `rollbar.rb`). Rake tasks that get run manually in production can end up sending errors to Rollbar: for example, providing the wrong arguments to a Rake task on the command line will print out an error to the user as well as report back to Rollbar (not necessary). I haven't dug into it that much, but I assume there has to be some environment variable provided by Rails that lets us know if we're in a Rake task or not

**Comment by ztmeyer** [comment]
I'm getting a "Permission denied" error when I try to push my branch, but here's one approach to consider in `Rollbar.rb`:

```ruby
  # Rollbar is only enabled in 'production' outside of rake tasks
  if !Rails.env.production? || /rake/ =~ File.split($PROGRAM_NAME).last
    config.enabled = false
  end
```
(More here: https://stackoverflow.com/questions/2467208/how-can-i-tell-if-rails-code-is-being-run-via-rake-or-script-generate)

This lets you disable Rollbar when a rake task is run by pattern matching on the `bin/rake` program name, which will be something like `/.../.rbenv/versions/2.4.2/bin/rake`.

There's one catch - the task `rake rollbar:test` will probably stop working. See this setting:
https://github.com/rollbar/rollbar-gem/blob/master/lib/rollbar/notifier.rb#L124

So if you use `rake rollbar:test` in production to verify your Rollbar config, this may not be the best approach. What are your thoughts?

**Comment by sman591** [comment]
@ztmeyer Aha, that'd do it! I think not being able to use `rake rollbar:test` is a fair tradeoff, as long as it's noted in the same `rollbar.rb` config (maybe a second right under the current comment).

For pushing your change, I think you might've tried to push directly to `codeRIT`'s origin rather than a fork. If you fork this repo (top right corner) you should be able to push your branch there, and make a pull request from it.

```bash
git remote add ztmeyer git@github.com:ztmeyer/brickhack.io.git
git push --set-upstream ztmeyer <your-branch-name>
```

**Comment by ztmeyer** [comment]
Oops - thanks for the instructions on fork/PR!

I've created a PR and linked it to this issue. Any comments welcome!

**Comment by sman591** [comment]
Fixed in #536 - forgot to close here. Thanks again @ztmeyer!
